pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -234,6 +234,32 @@ func (i *Instances) List(name_filter string) ([]string, error) {
 	return ret, nil
 }
 
+func getServerByHostName(client *gophercloud.ServiceClient, hostname string) (*servers.Server, error) {
+	// Depending on OpenStack setup, 'name' may include also domain name, while
+	// the server ID does not have it.
+
+	// Try with the full name first
+	server, err := getServerByName(client, hostname)
```

Comment:
Does it work if we look up by ^shortname(\.suffix)?$ rather than perform multiple lookups?